### PR TITLE
Fix defaults merge when get_all.

### PR DESCRIPTION
**Pull request by merqlove**
Subj. Select defaults, not fetch.
Otherwise starting_with selection can be wrong.
Also added another `defaults` fix with specs.

### CODE REVIEW

File: spec/rails-settings-cached/setting_spec.rb [hunk 1]
```diff
@@ -96,6 +105,11 @@
       expect(Setting.get_all).to include(:default1, :default2)
     end
 
+    it "should include namespace defaults" do
+      expect(RailsSettings::Default).to receive(:instance).and_return({ "test.default1" => 1, "test.default2" => '2', demo: 3 })
+      expect(Setting.get_all('test.')).to include(:'test.default1', :'test.default2')
```
Comment (huacnlee) [inline spec/rails-settings-cached/setting_spec.rb]: `Setting.get_all('test.')`

I don't want add this feature to let this gem more complex.

File: spec/rails-settings-cached/setting_spec.rb [hunk 2]
```diff
@@ -75,6 +75,15 @@
       it { expect(Setting['config.limit']).to eq 100 }
     end
 
+    context 'defaults within namespace for key' do
+      before do
+        allow(RailsSettings::Default).to receive(:instance).and_return({ "config.dcolor" => :blue, "config.dlimit" => 200 })
+      end
+
+      it { expect(Setting['config.dcolor']).to eq :blue }
+      it { expect(Setting['config.dlimit']).to eq 200 }
```
Comment (huacnlee) [inline spec/rails-settings-cached/setting_spec.rb]: Bug this is looks good, so would you please make an another PR?

File: spec/rails-settings-cached/setting_spec.rb (hunk 1 above)
Comment (merqlove) [inline spec/rails-settings-cached/setting_spec.rb]: This is more workaround than feature.
Currently this gem has wrong logic. For me it was broken (before this patch). 
In your version, we can select `{test: { fff: {} } }` data by 'test.fff', but we cannot store it same way (it will be stored as different path 'test.fff').
So, we must get it to work in useful way (as minimum) or it must have multi-direction storage/selection features ( as I offered).

File: spec/rails-settings-cached/setting_spec.rb (hunk 2 above)
Comment (merqlove) [inline spec/rails-settings-cached/setting_spec.rb]: Nice, just did it.

**Comment by merqlove** [comment]
@huacnlee, Thanks!)

**Comment by huacnlee** [comment]
Just released 0.6.5
